Fix empty configuration path that leads to the extension not working

MikaelMayer opened the pull request:
If someone deletes a configured language server runtime or compiler path, there will remain an empty string.

Previously, it would have been interpreted wrongly like a "relative path", and users would have needed to click on the gears icon next to the setting, and click "reset", which is unintuitive, since it was empty when they first edited it.

With this commit, it is correctly interpreted as "switch back to default"

This would have saved me the first time, and I just learned it would have saved someone else from spending a few hours of trouble.

Fixes #105

fabiomadge commented on src/language/dafnyInstallation.ts:
```diff
@@ -15,17 +15,24 @@ import Configuration from '../configuration';
+// Previously, it would have been interpreted wrongly like a "relative path".
+// Now it is interpreted as "switch back to default"
+function ifNullOrEmpty(a: string | null, b: string) {
+  return a === "" || a == null ? b : a;
```
```suggestion
  return a === '' || a == null ? b : a;
```

seebees commented:
I suspect that this will also fix #105.

keyboardDrummer commented on src/language/dafnyInstallation.ts:
```diff
@@ -15,17 +15,24 @@ import Configuration from '../configuration';
+// If someone deletes a configured language server runtime or compiler path, there will remain an empty string.
+// Previously, it would have been interpreted wrongly like a "relative path".
+// Now it is interpreted as "switch back to default"
+function ifNullOrEmpty(a: string | null, b: string) {
```
This seems like a strange function because `null` and `''` are both falsy in JavaScript, so instead of `isNullOrEmpty(a,b)`, I think you could also do `a || b`

MikaelMayer replied:
Good point. I'll even inline a || b since it's so simple.

keyboardDrummer reviewed (approved):
There's a new warning now: "Unexpected nullable string value in conditional. Please handle the nullish/empty cases explicitly". 

I guess eslint disagrees with my suggestion. Feel free to use the previous version you had since that it was it's asking you to do, or otherwise turn off the warning with `// eslint-disable-next-line`

MikaelMayer commented:
> // eslint-disable-next-line

I reverted to the previous one, but based on your suggestion, I simplified the conditional with `a ? a : b` instead of `a == ='' || a == null ? b : a`

camrein reviewed (approved):
> > // eslint-disable-next-line
> 
> I reverted to the previous one, but based on your suggestion, I simplified the conditional with `a ? a : b` instead of `a ==='' || a == null ? b : a`

I'm fine with both solutions. I tend to configure linters/static analyzers extremely strict and relax rules where needed. I suggest leaving the rule as is because the situation we've got here is rather seldom in my opinion. But an exception for this line is absolutely justified.
